spring-integration-file/src/test/java/org/springframework/integration/file/locking/FileChannelCacheTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.locking;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * @author Emmanuel Roux
+ */
+public class FileChannelCacheTests {
+
+	private File workdir;
+
+	@Rule
+	public TemporaryFolder temp = new TemporaryFolder() {
+
+		@Override
+		public void create() throws IOException {
```

Comment:
I don't see reason in such extra logic.
It looks like `temp.newFile("test0");` is fully enough against a default `public TemporaryFolder temp = new TemporaryFolder();`

Although I would insist on JUnit 5 already in the tests.
I'll back-port it to JUnit 4 for version `4.3.x` on merge

Reply:
Reason is that `TemporaryFolder.newFile(String)` calls `File.createNewFile()` which is not what is desired for test case of a non-existent file

Reply:
Well, you can do then `this.temp.getRoot()` for a `new File()`.
My point is: do not over-engineer a `TemporaryFolder` if we really can rely on its default public API.
With JUnit 5 we definitely don't have a luxury to extend extensions on the fly.

Reply:
You're right, let's go for that